test/test_replace_selections.py

```diff
@@ -0,0 +1,243 @@
+    arg_str: str = "foo"
+    arg_list: list = field(default_factory=lambda: [1, 2])
+    arg_dict: dict = field(default_factory=lambda: {"a": 1, "b": 2})
+    arg_union: str | Path = "./"
+    arg_tuple: tuple[int, int] = (1, 1)
+    arg_enum: Color = Color.BLUE
+    arg_dataclass: A = field(default_factory=A)
+    arg_subgroups: A | B = subgroups(
+        {
+            "a": A,
+            "a_1.23": functools.partial(A, a=1.23),
+            "b": B,
+            "b_bob": functools.partial(B, b="bob"),
+        },
+        default="a",
+    )
+    arg_optional: A | None = None
+    arg_optional_1: A | None = None
+    arg_union_dataclass: A | B = field(default_factory=A)
+    arg_union_dataclass_init_false: A | B = field(init=False)
+
+    def __post_init__(self):
+        self.arg_union_dataclass_init_false = copy.copy(self.arg_union_dataclass)
+
+
+@pytest.mark.parametrize(
+    ("start", "changes", "selections", "expected"),
+    [
+        (
+            AB(),
+            {"a_or_b": {"a": 1.0}},
+            None,
+            AB(a_or_b=A(a=1.0)),
+        ),
+        (AB(a_or_b=B()), {"a_or_b": {"b": "foo"}}, None, AB(a_or_b=B(b="foo"))),
+        (AB(), {"a_or_b": {"b": "foo"}}, {"a_or_b": "b"}, AB(a_or_b=B(b="foo"))),
+        (
+            AB(),
+            {"a_or_b": B(b="bob")},
+            None,
+            AB(a_or_b=B(b="bob")),
+        ),
+        (
+            NestedSubgroupsConfig(ab_or_cd=AB(a_or_b=B())),
+            {"ab_or_cd": {"integer_in_string": "2", "a_or_b": {"b": "bob"}}},
+            None,
+            NestedSubgroupsConfig(ab_or_cd=AB(integer_in_string="2", a_or_b=B(b="bob"))),
+        ),
+        (
+            NestedSubgroupsConfig(ab_or_cd=AB(a_or_b=B())),
+            {"ab_or_cd.integer_in_string": "2", "ab_or_cd.a_or_b.b": "bob"},
+            None,
+            NestedSubgroupsConfig(ab_or_cd=AB(integer_in_string="2", a_or_b=B(b="bob"))),
+        ),
+        (
+            NestedSubgroupsConfig(),
+            {"ab_or_cd.integer_in_string": "2", "ab_or_cd.a_or_b.b": "bob"},
+            {"ab_or_cd": "ab", "ab_or_cd.a_or_b": "b"},
+            NestedSubgroupsConfig(ab_or_cd=AB(integer_in_string="2", a_or_b=B(b="bob"))),
+        ),
+        (
+            NestedSubgroupsConfig(),
+            None,
+            {"ab_or_cd": "cd", "ab_or_cd.c_or_d": "d"},
+            NestedSubgroupsConfig(ab_or_cd=CD(c_or_d=D())),
+        ),
+        (
+            NestedSubgroupsConfig(),
+            {"ab_or_cd.c_or_d.d": 1},
+            {"ab_or_cd": "cd", "ab_or_cd.c_or_d": "d"},
+            NestedSubgroupsConfig(ab_or_cd=CD(c_or_d=D(d=1))),
+        ),
+        (
+            AllTypes(),
+            {"arg_subgroups.b": "foo", "arg_optional.a": 1.0},
+            {"arg_subgroups": "b", "arg_optional": A},
+            AllTypes(arg_subgroups=B(b="foo"), arg_optional=A(a=1.0)),
+        ),
+    ],
+)
+@pytest.mark.parametrize("pass_dict_as_kwargs", [True, False])
+def test_replace_selections(
```

Comment:
I think it's important that we test `replace_selected_dataclass` first, by itself.

Also, let's avoid using too many dataclasses in the tests. they are becoming pretty hard to read.

Reply:
Keep only replace_subgroups function in this updating.